test/amqp/client_test.rb

```diff
@@ -275,6 +275,40 @@ def test_it_can_reject
     assert_equal 2, i
   end
 
+  def test_it_can_recover
```

Comment:
also make a test with recover with requeue, then the same consumer should get the unacked messages again